shared/src/main/kotlin/com/egm/stellio/shared/util/NgsiLdParsingUtils.kt

```diff
@@ -194,8 +199,10 @@ object NgsiLdParsingUtils {
                         NGSILD_TIME_TYPE -> LocalTime.parse(finalValue)
                         else -> firstListEntry["@value"]
                     }
-                } else {
+                } else if (firstListEntry["@value"] != null) {
                     firstListEntry["@value"]
+                } else {
```

Comment:
why did you have to do that?

Reply:
Since it is expanded with the following format:
```
{ ......
    "https://uri.etsi.org/ngsi-ld/datasetId" = [{
        "@id"= dataset_id_value
    }]
 ......
}
```

Reply:
add it as a comment in the code